tests/functional/test_post_company.py

```diff
@@ -0,0 +1,114 @@
+from flask import json
+
+
+def test_post_valid_company(client, get_token):
+    '''
+    GIVEN a Flask application configured for testing
+    WHEN a POST request is made to /api/v1/companies with a valid company
+    THEN check that the response is valid and particular data can be found
+    '''
+    data = {
+        "branches": [
+            "Nog niet bestaande branche"
+        ],
+        "city_name": "Lutjebroek",
+        "company_name": "Test company #999",
+        "company_size": "11-50",
+        "disciplines": [
+            "Conceptontwikkeling",
+            "Conversie-optimalisatie",
+            "Strategie",
+            "Nog niet bestaande discipline",
+            "Nog niet bestaande discipline #2"
+        ],
+        "logo_image_src": "https://www.nos.nl/",
+        "region": "Remote",
+        "tags": [
+            "Angular",
+            "App-bouwer",
+            "Nog niet bestaande tag",
+            "Nog niet bestaande tag #2"
+        ],
+        "website": "https://www.google.com/",
+        "year": 1969
+    }
+
+    response = client.post('/api/v1/companies', data=json.dumps(data),
+                           headers={'Content-Type': 'application/json', 'Authorization': 'Bearer {}'.format(get_token)},)
+
+    result = json.loads(response.get_data(as_text=True))
+
+    assert response.status_code == 201
+    assert result['company_id'] == 1
+    assert result['company_name'] == 'Test company #999'
+    assert result['city_name'] == 'Lutjebroek'
+    assert result['region'] == 'Remote'
+    assert result['company_size'] == '11-50'
+    assert result['branches'][0] == 'Nog niet bestaande branche'
+    assert result['website'] == 'https://www.google.com/'
+    assert result['year'] == 1969
+
+
+def test_post_invalid_company(client, insert_data_db, get_token):
```

Comment:
This seems like 3 tests in one ..

Perhaps split up like: 

```
GIVEN a Flask application configured for testing
WHEN a POST request is made to /api/v1/companies with a company_id in the body
THEN the app should respond with a 400 and an error message
```

```
GIVEN a Flask application configured for testing
WHEN a POST request is made to /api/v1/companies without all the required fields
THEN the app should respond with a 400 and an error messages for each missing field
```

etc ...

Reply:
It is typical to have a *lot* more unhappy paths than happy paths when it comes to testing

Alternatively we could `unittest` the CompaniesSchema and check if the validation work according to plan
These tests would run faster as we don't have as much setup / teardown, request / response

Reply:
Agree, these are just 3 tests cramped in 1. Splitting them up will surely help readability and clearity.

Reply:
Regarding the unittests on the validation schema's, I think we should include these thoughts in the PR #8 and consider using this to speed up the testing.